### ISSUE TITLE
search does not support utf8

### ISSUE THREAD

**Issue by Antoine-35** [squidfunk/mkdocs-material#2970]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [ ] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When using the search, utf-8 special characters are not displayed correctly.
This only happens in the display of the different search responses.
I don't have this feature with RTD

For exemple a search for "Source & Destination"

This happens for all utf8 encoded characters.

Example of a text with the problem:
```## Format

### Source & Destination

![Capture exemple sources](img/02_exemples_sources.png){ align=right }

Elles peuvent etre dans plusieurs formats:

* IP
* Subnet/masque
* Range IP: IP1-IP2
* FQDN
* Objects depuis un firewall


```

### Expected behaviour

The quick search displays "Source & Destination".

### Actual behaviour

![image](https://user-images.githubusercontent.com/52744345/130627842-a282b347-526f-46b5-a5ef-b5363eb3ab23.png)

### Steps to reproduce

1. Create page with some utf8 text: ' & 
2. search for this text

### Package versions

- Python: `python --version` 3.6.9
- MkDocs: `mkdocs --version` 1.2.2
- Material: `pip show mkdocs-material | grep -E ^Version` Version: 7.2.0

### Configuration

```yaml
site_name: Documentation Tufin
site_url: ****

theme:
  name: material
  features:
    - navigation.top
    - navigation.tabs
    - search.highlight
    - search.suggest
    - search.share
  plugins:
    - search
  language: fr
  palette:
    - media: "(prefers-color-scheme: light)"
      scheme: default
      toggle:
        icon: material/toggle-switch-off-outline
        name: Switch to dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/toggle-switch
        name: Switch to light mode
markdown_extensions:
  - attr_list
  - admonition
```

### System information

- Operating system: W10
- Browser: Chrome and Firefox

**Comment by squidfunk** [comment]
Thanks for reporting. Regression due to #2879

**Comment by squidfunk** [comment]
Fixed in 829820a23.

**Comment by squidfunk** [comment]
Released as part of 7.2.5.

**Comment by Antoine-35** [comment]
I confirm that it is ok, thank you
